src/meta/src/hummock/manager/mod.rs

```diff
@@ -493,3 +501,68 @@ async fn write_exclusive_cluster_id(
         }
     }
 }
+
+#[derive(Debug, Clone)]
+pub struct TableWriteThroughputStatistic {
+    pub throughput: u64,
+    pub timestamp: i64,
+}
+
+#[derive(Debug, Clone)]
+pub struct TableWriteThroughputStatisticManager {
+    table_throughput: HashMap<u32, Vec<TableWriteThroughputStatistic>>,
+    max_statistic_expired_time: i64,
+}
+
+impl TableWriteThroughputStatisticManager {
+    pub fn new(max_statistic_expired_time: i64) -> Self {
+        Self {
+            table_throughput: HashMap::new(),
+            max_statistic_expired_time,
+        }
+    }
+
+    pub fn add_table_throughput_with_ts(&mut self, table_id: u32, throughput: u64, timestamp: i64) {
+        let table_throughput = self
+            .table_throughput
+            .entry(table_id)
+            .or_insert_with(Vec::new);
+        table_throughput.push(TableWriteThroughputStatistic {
+            throughput,
+            timestamp,
+        });
+
+        table_throughput
+            .retain(|statistic| timestamp - statistic.timestamp <= self.max_statistic_expired_time);
+    }
+
+    pub fn get_table_throughput(
+        &self,
+        table_id: u32,
+        expired_time: i64,
+    ) -> Option<Vec<&TableWriteThroughputStatistic>> {
+        let timestamp = chrono::Utc::now().timestamp();
+        if let Some(statistics) = self.table_throughput.get(&table_id) {
+            let table_throughput_statistics = statistics
+                .iter()
+                .filter(|statistic| timestamp - statistic.timestamp <= expired_time)
+                .collect_vec();
+
+            if !table_throughput_statistics.is_empty() {
+                Some(table_throughput_statistics)
+            } else {
+                None
+            }
+        } else {
+            None
+        }
+    }
+
+    pub fn retain(&mut self) {
+        self.table_throughput.retain(|_, v| !v.is_empty());
+    }
```

Comment:
Given that we only remove items in `table_throughput` via `add_table_throughput_with_ts`, we don't need this separate `retain` method and we can just do it in place in `add_table_throughput_with_ts`

Reply:
Actually, I think we need to way to remove dropped table in TableWriteThroughputStatisticManager.

Reply:
remove the `retain ` method and trigger the `remove` when all statistics are expired

Reply:
When do we remove the items for dropped table? I still cannot find the relevant code path.